crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -248,6 +248,26 @@ impl Settlement {
             }
         };
 
+        {
+            // For testing purposes, always calculate CIP38 score, even if the score is not
+            // used.
+            let trades = self
+                .solutions
+                .iter()
+                .filter_map(|(id, solution)| match solution.scorable_trades() {
+                    Ok(trades) => Some(trades),
+                    Err(err) => {
+                        tracing::debug!(?id, ?err, "could not prepare scorable settlement");
+                        None
+                    }
+                })
+                .flatten()
+                .collect();
+            let settlement = competition::settled::Settlement::new(trades);
+            let score = settlement.score(&auction.normalized_prices());
+            tracing::info!(?score, "CIP38 score for settlement: {:?}", self.solutions());
+        }
```

Comment:
I still think the final score value should simply be exposed on solution. 

Either we pass in the auction to `Solution::new` and create the CIP38 score there (creating a new variant on competition::SolverScore) or we create a new method to compute the score which takes the auction as an argument. 
The latter has the issue of finding a good name (given that `score()` is already taken). I think I'm getting more accustomed to the idea of adding a temporary method `surplus_score(auction: &Auction)` which gets turned into `score()` once we get rid of the legacy scoring options.

Reply:
refactored.